awx/api/generics.py

```diff
@@ -269,12 +270,19 @@ def get_serializer(self, *args, **kwargs):
         return serializer
 
     def get_queryset(self):
-        #if hasattr(self.request.user, 'get_queryset'):
-        #    return self.request.user.get_queryset(self.model)
         if self.queryset is not None:
             return self.queryset._clone()
         elif self.model is not None:
-            return self.model._default_manager.all()
+            qs = self.model._default_manager
+            try:
+                access_class = access_registry[self.model]
+                if access_class.select_related:
+                    qs = qs.select_related(*access_class.select_related)
+                if access_class.prefetch_related:
+                    qs = qs.prefetch_related(*access_class.prefetch_related)
+            except KeyError:
+                pass  # Model doesn't have a registered access class, ok
+            return qs
```

Comment:
this feels clunky. A much more simple approach might be to simply move the ListView `get_queryset` to the common parent class. That could have some interesting consequences (like forcing a 404 when navigating to something you don't have permissions to). I'm not partial to either approach.

https://github.com/AlanCoding/awx-1/blob/41e0f833a7d76ebc0ef9409510e31ea22528b30a/awx/api/generics.py#L323-L324